### PR TITLE
Update query editor colorization to match SSMS

### PR DESCRIPTION
**Pull request by corivera**
Includes updated colorization for the following:
- local variables (tokens prepended with "@")
- global system variables (tokens prepended with "@@")
- unicode string literal prefix ("N" in front of strings)
- bracketed identifiers (tokens surrounded by square brackets)
- builtin SQL Server functions

Fixes: https://github.com/Microsoft/sqlopsstudio/issues/345 https://github.com/Microsoft/sqlopsstudio/issues/225

### CODE REVIEW

File: extensions/sql/syntaxes/SQL.plist [hunk 1]
```diff
@@ -15,10 +15,16 @@
...
-      			<key>name</key>
-      			<string>keyword.other.sql</string>
+			<key>match</key>
+			<string>(((?<!@)@)|(\[\s*))\b(?i)(\w+)\b</string>
+			<key>name</key>
+			<string>text</string>
```
Comment (corivera) [inline extensions/sql/syntaxes/SQL.plist]: Anything with a leading '@' or '[' should be treated as regular, uncolored text. Is there a better "category" to use here? There's no color theming for "text", so it accomplishes the same goal.

**Comment by corivera** [comment]
Added colorization for FULL OUTER JOIN. Fixes https://github.com/Microsoft/sqlopsstudio/issues/584

File: extensions/sql/syntaxes/sql.tmLanguage.json [hunk 2]
```diff
@@ -14,7 +14,11 @@
...
...t|start_date|started|stat_header|state|statement|static|statistics|statistics_incremental|statistics_norecompute|statistics_only|statman|stats_stream|status|stop|stop_on_error|stopat|stopatmark|stopbeforemark|stoplist|stopped|string_delimiter|subject|supplemental_logging|supported|suspend|symmetric|synchronous_commit|synonym|sysname|system|system_time|system_versioning|table|tableresults|tablock|tablockx|take|tape|target|target_index|target_partition|tcp|temporal_history_retention|text|textimage_on|then|thesaurus|throw|time|timeout|timestamp|tinyint|to|top|torn_page_detection|track_columns_updated|tran|transaction|transfer|triple_des|triple_des_3key|trustworthy|try|tsql|type|type_desc|type_warning|tzoffset|uid|unbounded|uncommitted|uniqueidentifier|unlimited|unload|unlock|unsafe|updlock|url|use|useplan|useroptions|use_type_default|using|utcdatetime|valid_xml|validation|value|values|varbinary|varchar|verbose|verifyonly|version|view_metadata|virtual_device|visiblity|waitfor|webmethod|weekday|weight|well_formed_xml|when|while|widechar|widechar_ansi|widenative|windows|with|within|witness|without|without_array_wrapper|workload|wsdl|xact_abort|xlock|xml|xmlschema|xquery|xsinil|zone)\\b",
```
Comment (kburtram) [inline extensions/sql/syntaxes/sql.tmLanguage.json]: do you know why we need to repeat the whole keyword list?

**Comment by abist** [comment]
Also fixes https://github.com/Microsoft/sqlopsstudio/issues/557 it seems

**Comment by corivera** [comment]
@abist That issue will get fixed in my next commit

**Pull request merged**